components/renku_data_services/crc/models.py

```diff
@@ -390,3 +413,68 @@ def get_default_resource_class(self) -> ResourceClass | None:
             if rc.default:
                 return rc
         return None
+
+
+class RemoteConfigurationKind(StrEnum):
+    """Remote resource pool kinds."""
+
+    firecrest = "firecrest"
+
+
+@dataclass(frozen=True, eq=True, kw_only=True)
+class RemoteConfigurationFirecrest:
+    """Model for remote configurations using the FirecREST API."""
+
+    kind: RemoteConfigurationKind = RemoteConfigurationKind.firecrest
+    provider_id: str | None = None
+    api_url: str
+    system_name: str
+    partition: str | None = None
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Self:
+        """Convert a dict object into a RemoteConfiguration instance."""
+        kind = data.get("kind")
+        if kind == RemoteConfigurationKind.firecrest.value:
+            return cls(
+                kind=RemoteConfigurationKind.firecrest,
+                provider_id=data.get("provider_id") or None,
+                api_url=data["api_url"],
+                system_name=data["system_name"],
+                partition=data.get("partition") or None,
+            )
+        raise errors.ValidationError(message=f"Invalid kind for remote configuration: '{kind}'")
+
+    def to_dict(self) -> dict[str, Any]:
+        """Convert this instance of RemoteConfiguration into a dictionary."""
+        res = asdict(self)
+        res["kind"] = self.kind.value
+        return res
+
+
+@dataclass(frozen=True, eq=True, kw_only=True)
+class RemoteConfigurationPatchReset:
+    """Model used to unset a remote configuration."""
+
+    pass
```

Comment:
You dont need this. We already have a dedicated type such a case. Check `RESET` in `components/renku_data_services/base_models/core.py`. Instead of making this custom type just pass in this RESET object and use that.

Reply:
Done in 68db9ef0.